service_crategen/src/commands/generate/codegen/query.rs

```diff
@@ -94,15 +94,14 @@ impl GenerateProtocol for QueryGenerator {
                 }}
 
 
-                async fn sign_and_dispatch<E>(
+                async fn sign_and_dispatch(
                     &self,
                     request: SignedRequest,
-                    from_response: fn (BufferedHttpResponse) -> RusotoError<E>,
-                ) -> Result<HttpResponse, RusotoError<E>> {{
+                ) -> Result<HttpResponse, RusotoError<std::convert::Infallible>> {{
                     let mut response = self.client.sign_and_dispatch(request).await?;
                     if !response.status.is_success() {{
-                        let response = response.buffer().await.map_err(RusotoError::HttpDispatch)?;
```

Comment:
What is the rationale for changing the returned variant from `HttpDispatch` to `Unknown` here? I'm imagine this might break a few clients, and I would think `HttpDispatch` is more appropriate -- but feel free to set me straight here.

Reply:
I think it should still be `HttpDispatch` though? It just gets wrapped by the `From` impl instead?